core/src/main/java/tc/oc/pgm/projectile/ProjectileModule.java

```diff
@@ -90,13 +96,53 @@ public ProjectileModule parse(MapFactory factory, Logger logger, Document doc)
             coolDown,
             throwable,
             precise,
-            blockMaterial);
+            blockMaterial,
+            scale,
+            solidBlockCollision,
+            maxTravelTime);
 
         factory.getFeatures().addFeature(projectileElement, projectileDefinition);
         projectiles.add(projectileDefinition);
       }
 
       return projectiles.isEmpty() ? null : new ProjectileModule(ImmutableSet.copyOf(projectiles));
     }
+
+    private static final Map<String, ProjectileDefinition.ProjectileEntity.AbstractEntity> ABSTRACT_ENTITY_MAP =
+        ImmutableMap.of(
+          "block", new ProjectileDefinition.ProjectileEntity.AbstractEntity(
+                  ProjectileDefinition.ProjectileEntity.AbstractEntityType.BLOCK
+              )
+        );
+
+    private static ProjectileDefinition.ProjectileEntity parseProjectileEntity(
+      final Element element,
+      final String attributeName,
+      final Class<? extends Entity> def
+    ) throws InvalidXMLException {
```

Comment:
```suggestion
    private static ProjectileDefinition.ProjectileEntity parseProjectileEntity(XmlFluentParser parser, Element el) throws InvalidXMLException {
```
Attribute name is *always* "projectile" and default entity is *always* arrow, there's no need to make them parameters, instead pass in the parser which will be of use